### ISSUE TITLE
Complete the CompileCommands.md

### ISSUE THREAD

**Issue by theofficialgman** [Pi-Apps-Coders/files#32]
this doesn't need to be verbatim exactly how to build each app or look pretty, but it would be nice to at least semi-document all apps here and how to go about reproducing their binarires, debs, or tarfiles (whatever is relavent).

for example, I'm looking at alacricity terminal and would like to make a deb myself but don't know if this was done with some automated method or if it was manually done.

**Comment by Botspot** [comment]
I am interested in seeing this happen too.
Compile commands are both useful and reduce the bus factor in the event that a maintainer disappears for good.

**Comment by theofficialgman** [comment]
github desktop is also undocumented but I will add that myself if I am given push access to this repo.

**Comment by theofficialgman** [comment]
> I thought you already had push access, as you are a member of Pi-Apps-Coders.

I'm a member, but that didn't give me any other privileges besides seeing who the other members are

**Comment by Botspot** [comment]
> > I thought you already had push access, as you are a member of Pi-Apps-Coders.
> 
> I'm a member, but that didn't give me any other privileges besides seeing who the other members are

Try it again. I've enabled write permissions for all members of this organization. (I thought this was already turned on, sorry about that)

**Comment by theofficialgman** [comment]
> Try it again. I've enabled write permissions for all members of this organization. (I thought this was already turned on, sorry about that)

I have access now, thanks

**Comment by theofficialgman** [comment]
https://github.com/alacritty/alacritty/pull/5902

given that upstream now how GLES2 support, we should build debs on upstream alacritty master (the latest release it too old to include it)

in the meantime, I corrected the current deb for arm64 to have libc 2.18 in the control file instead of 2.28, since I did an objdump -T and found nothing higher to be requried

**Comment by theofficialgman** [comment]
> * LightPad: I think the debs are taken from the [latest release](https://github.com/libredeb/lightpad/releases/tag/v0.0.8) but I cannot confirm that for sure, if they are then I suggest we remove them from thisrepo since they are not needed

I diffed the debs, they are named differently and have binary differences

@Itai-Nelken what did you change for lightpad from the official release if you remember? https://github.com/Pi-Apps-Coders/files/commit/7e8781523f20f6eee728411f6da16c3f225c2a21

**Comment by theofficialgman** [comment]
oh I see.. maybe it came from https://github.com/Botspot/pi-apps/commit/c6ce6596f0f05f61b21bd2ec12976f161664d345

@scamdotnet can you elaborate what you changed?

**Comment by scamdotnet** [comment]
I'm honestly not sure as I can't download the deb file anymore. I might have a copy of it on one of my Pi's, but I would have to set them up. I'll keep you updated if I can find the new deb file though!

**Comment by scamdotnet** [comment]
Quick question, is the file hosted at https://github.com/Pi-Apps-Coders/files/raw/main/lightpad_0.0.8rev1_arm64.deb a reupload of http://transfer.sh/get/1HvLYEm/lightpad_0.0.8~rev1_arm64.deb?

**Comment by scamdotnet** [comment]
Okay so from comparing the one from the original URL before I made my commit (https://github.com/Botspot/pi-apps/commit/c6ce6596f0f05f61b21bd2ec12976f161664d345), with the one from this repo, it appears that the Debian control file was changed from requiring libc6 (>= 2.29), to requiring libc6 (>= 2.17), as well as fixing a spelling mistake in the control file (Writen in the Original, Written in the one in this repo). Looking at the md5sums, the binary file was modified, not sure how, and so was the changelog. Comparing the changelogs, it looks like the modified one is actually a slightly earlier version, as it is rev1, while the one in the official repo is rev2. The only difference in the changelog besides the header for the version is a note saying that rev2 is a rebuild for groovy.

TLDR: the only real change that I believe I made was reducing the requirement for libc6 from >= 2.29, to >= 2.17. I think this was because this caused some kind of conflict in the raspbian repositories, although I can't be sure.

**Comment by theofficialgman** [comment]
> TLDR: the only real change that I believe I made was reducing the requirement for libc6 from >= 2.29, to >= 2.17. I think this was because this caused some kind of conflict in the raspbian repositories, although I can't be sure.

I see.. it looks like the official release has a reupload for the arm64 file (I assume thats what you were looking at). that would explain the control file differences and binary differences. regardless, the official builds will not work due to being built on groovy so this is good information, we will keep the debs hosted here then.

you can see the official release deb upload dates before where the arm64 version was reuploaded later
```
[lightpad_0.0.8.rev1_amd64.deb](https://github.com/libredeb/lightpad/releases/download/v0.0.8/lightpad_0.0.8.rev1_amd64.deb)
30.9 KB
Jul 22, 2020
[lightpad_0.0.8.rev1_arm64.deb](https://github.com/libredeb/lightpad/releases/download/v0.0.8/lightpad_0.0.8.rev1_arm64.deb)
29.5 KB
Jan 06, 2021
[lightpad_0.0.8.rev1_armhf.deb](https://github.com/libredeb/lightpad/releases/download/v0.0.8/lightpad_0.0.8.rev1_armhf.deb)
27.6 KB
Jul 22, 2020
[lightpad_0.0.8.rev1_i386.deb](https://github.com/libredeb/lightpad/releases/download/v0.0.8/lightpad_0.0.8.rev1_i386.deb)
34.1 KB
Jul 22, 2020
```

**Comment by scamdotnet** [comment]
Glad I could help! If there is anything else you think I could help out with, feel free to at me on github.
